ovs/lib/mdsservice.py

```diff
@@ -72,6 +72,9 @@ def prepare_mds_service(storagerouter, vpool, fresh_only):
         """
         from ovs.lib.storagedriver import StorageDriverController
 
+        prepare_mds_lock = volatile_mutex('prepare_mds_{0}'.format(storagerouter.guid))
+        prepare_mds_lock.acquire(30)
```

Comment:
This must be in a try/finally. Otherwise it might be possible that the lock isn't released.